includes/configuration_managers/class-woocommerce-configuration-manager.php

```diff
@@ -170,21 +218,23 @@ protected function create_donation_product( $args ) {
 	/**
 	 * Update the donations products.
 	 *
-	 * @param string $args Donations settings. See $defaults at top of this method for more info.
+	 * @param string $args Donations settings.
 	 */
 	protected function update_donation_product( $args ) {
-		$defaults = [
-			'name' => '',
-			'suggestedAmount' => 0,
-			'imageID' => 0,
-		];
+		$defaults = $this->get_donation_default_settings();
 		$args = wp_parse_args( $args, $defaults );
 		$product_id = get_option( self::DONATION_PRODUCT_ID_OPTION, 0 );
 		$parent_product = \wc_get_product( $product_id );
```

Comment:
I've been taking a slightly different approach to what goes in the configuration manager vs. the wizard class. Any statement that contains the plugin's classname goes in the configuration manager (same for setting plugin options) but everything else goes in the wizard class. I'm honestly not sure if that approach breaks down here, with such a mix of WC methods and utilities, but there might be some ways to separate the concerns. My favorite one so far is `Newspack Ads` — even though we control both sides of the equation the config manager methods are ultra simple: https://github.com/Automattic/newspack-plugin/blob/master/includes/configuration_managers/class-newspack-ads-configuration-manager.php. Part of the rationale for this is that these configuration manager classes are going to be the most brittle part of Newspack—prone to breakage whenever a plugin update is released. Even though we're not doing it yet, I'd like to build a robust suite of tests that download the latest version of each plugin and run though all config manager methods. This will be an effective way to quickly catch issues when something changes in the 3rd party plugin, and the simpler the config manager methods are the cleaner the tests will be.

Reply:
In the interest of not getting too lost in the weeds, I'll leave it for now, and we can refactor in the future. I like the idea of having a test suite for our configuration managers to catch issues when plugins update; those are definitely going to be our most fragile classes.